src/crypto/sha256.cpp

```diff
@@ -14,6 +14,41 @@
 #include <intel-ipsec-mb.h>
 #endif
 
+#if defined(__arm__) || defined(__aarch32__) || defined(__arm64__) || defined(__aarch64__) || defined(_M_ARM)
+# if defined(__GNUC__)
+#  include <stdint.h>
+# endif
+# if defined(__ARM_NEON) || defined(_MSC_VER) || defined(__GNUC__)
+#  include <arm_neon.h>
+# endif
+/* GCC and LLVM Clang, but not Apple Clang */
+# if defined(__GNUC__) && !defined(__apple_build_version__)
+#  if defined(__ARM_ACLE) || defined(__ARM_FEATURE_CRYPTO)
+#   include <arm_acle.h>
+#  endif
+# endif
+#endif  /* ARM Headers */
+
+static const uint32_t K[] =
```

Comment:
potentially yes, it could be referenced in the legacy implementation of Transform instead of hardcoded values..
`Round(a, b, c, d, e, f, g, h, 0x428a2f98, w0 = ReadBE32(chunk + 0));`

i'd probably make a changes to use round constants in the old sha1 and sha512 too
or I could simply move it in the ARM guard, whichever is preferable

Reply:
Let's leave it for now then, and make a decision on that later - either through further use or by restricting it to the functions that use it otherwise.